mesonbuild/modules/fs.py

```diff
@@ -60,69 +59,57 @@ def _resolve_dir(self, state: 'ModuleState', arg: str) -> Path:
         return path
 
     def _check(self, check: str, state: 'ModuleState', args: T.Sequence[str]) -> ModuleReturnValue:
-        if len(args) != 1:
-            raise MesonException('fs.{} takes exactly one argument.'.format(check))
         test_file = self._resolve_dir(state, args[0])
         val = getattr(test_file, check)()
         if isinstance(val, Path):
             val = str(val)
         return ModuleReturnValue(val, [])
 
-    @stringArgs
     @noKwargs
     @FeatureNew('fs.expanduser', '0.54.0')
-    def expanduser(self, state: 'ModuleState', args: T.Sequence[str], kwargs: dict) -> ModuleReturnValue:
-        if len(args) != 1:
-            raise MesonException('fs.expanduser takes exactly one argument.')
-        return ModuleReturnValue(str(Path(args[0]).expanduser()), [])
+    @typed_pos_args('fs.expanduser', str)
+    def expanduser(self, state: 'ModuleState', args: T.Tuple[str], kwargs: T.Dict[str, T.Any]) -> ModuleReturnValue:
+        return ModuleReturnValue(os.path.expanduser(args[0]), [])
```

Comment:
I thought we wanted to use pathlib everywhere?

Reply:
Ok forget me, you explain in the commit message. I don't think performance matter in this case, but it's indeed not worth doing conversion from/to str.